File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int l=0;
    
        ListNode temp=head;
        while(temp!=null){
            l++;
            temp=temp.next;
        }
        int toremove=l-n;
        temp=head;
        int c=0;
     if(toremove==0){
        return head.next;
     }
        while(temp!=null){
            c++;
            if(c==toremove){
                temp.next=temp.next.next;
                break;
            }
            temp=temp.next;
        }
        return head;

    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        temp=head
        l=0
        while temp:
            l+=1
            temp=temp.next
        toremove=l-n
        if toremove==0:
            return head.next
        temp=head
        c=0
        while temp:
            c+=1
            if c==toremove:
                temp.next=temp.next.next
                break
            temp=temp.next
        return head


        